uwblueprint/focus-on-nature
# feat: create get camp filter

## Notion ticket link

[Get Camp Filter](https://www.notion.so/uwblueprintexecs/Tasks-38b49f83f6ce4b07ae188dabb37631e8?p=d40365fe29b34a71a3358ffe7d06afcd)



## Implementation description
* get camps endpoint can take in optional parameters: `campYear` and `campStatus`, and returns all camps where campSession of each camp is filtered by `campYear` and `campStatus`
* added camp year and camp status validators



## Steps to test
1. Camp year should be of YYYY format, and campStatus should be one of Active, Published, Draft, Archived
2. After making a GET request to say `http://localhost:5000/camp?campStatus=Active`, the response body should contain all the camps where campSessions in each camp could be empty if none of the camp session have status of "Active"
3. Similarly, make a GET request to `http://localhost:5000/camp?campYear=2021` and verify response body
4. Combine both the parameters: make a GET request to `http://localhost:5000/camp?campStatus=Active&campYear=2021`
5. Verify the error messages received when passing in invalid camp status/year



## What should reviewers focus on?
* Verifying if response body contains all the desired fields


## Checklist
